src/communication/network_interface_sr.c

```diff
@@ -4674,6 +4674,55 @@ shnsw_delete_index (THREAD_ENTRY * thread_p, unsigned int rid, char *request, in
   css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
 }
 
+void
+shnsw_load_index (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
+{
+  BTID btid;
+  BTID *return_btid = NULL;
+  int dimension, hnsw_M, hnsw_efConstruction, metric_type;
+  OID *class_oids = NULL;
+  HFID *hfids = NULL;
+  int n_classes, n_attrs, *attr_ids = NULL;
+  OR_ALIGNED_BUF (OR_INT_SIZE + OR_BTID_ALIGNED_SIZE) a_reply;
+  char *reply = OR_ALIGNED_BUF_START (a_reply);
+  char *ptr;
+
+  ptr = or_unpack_btid (request, &btid);
+  ptr = or_unpack_int (ptr, &n_classes);
+  ptr = or_unpack_int (ptr, &n_attrs);
+  ptr = or_unpack_int (ptr, &dimension);
+  ptr = or_unpack_int (ptr, &hnsw_M);
+  ptr = or_unpack_int (ptr, &hnsw_efConstruction);
+  ptr = or_unpack_int (ptr, &metric_type);
+
+  ptr = or_unpack_oid_array (ptr, n_classes, &class_oids);
+  if (ptr == NULL)
+    {
+      (void) return_error_to_client (thread_p, rid);
+      return;
+    }
+  ptr = or_unpack_int_array (ptr, (n_classes * n_attrs), &attr_ids);
+  if (ptr == NULL)
+    {
+      (void) return_error_to_client (thread_p, rid);
+      return;
+    }
+  ptr = or_unpack_hfid_array (ptr, n_classes, &hfids);
+  if (ptr == NULL)
+    {
+      (void) return_error_to_client (thread_p, rid);
+      return;
+    }
+
+  return_btid =
+    xhnsw_load_index (thread_p, &btid, class_oids, n_classes, n_attrs, attr_ids, hfids, dimension, hnsw_M,
+		      hnsw_efConstruction, metric_type);
+
+  ptr = or_pack_int (reply, NO_ERROR);
+  ptr = or_pack_btid (ptr, return_btid);
+  css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
```

Comment:
```suggestion
  ptr = or_pack_int (reply, NO_ERROR);
  ptr = or_pack_btid (ptr, return_btid);
  css_send_data_to_client (thread_p->conn_entry, rid, reply, OR_ALIGNED_BUF_SIZE (a_reply));
  
    if (attr_ids != NULL)
    {
      db_private_free_and_init (thread_p, attr_ids);
    }
  if (class_oids != NULL)
    {
      db_private_free_and_init (thread_p, class_oids);
    }
  if (hfids != NULL)
    {
      db_private_free_and_init (thread_p, hfids);
    }

```

debug 모드에서 위 패치를 통해 vector index 를 생성할 시 restract_assert(false) 문제를 해결할 수 있는 것으로 보이는데, 확인 부탁드립니다.

Reply:
@hgryoo 께서 알려주신 서버 로그의 내용입니다.

아래 내용에 따르면
or_unpack_hfid_array
or_unpack_int_array
or_unpack_oid_array

함수 사용으로 인해 발생하는 것으로 추정됩니다.

```
Time: 08/12/25 22:13:35.669 - DEBUG *** file /home/vimkim/gh/cb/load-vector-index/src/storage/hnsw_usearch.cpp, line 486
Added element with OID 1103806660609 to HNSW Index ID 0.
Time: 08/12/25 22:13:35.669 - DEBUG *** file /home/vimkim/gh/cb/load-vector-index/src/base/resource_tracker.cpp, line 79

   +--- Virtual Memory
         +--- amount = 2 (threshold = 32767)
         +--- stack_level = 0
            +--- tracked res_ptr=0x302f500 amount=1 | first_caller=object_representati:1586
            +--- tracked res_ptr=0x302f520 amount=1 | first_caller=object_representati:1759
            +--- tracked res_ptr count = 2
```